### ISSUE TITLE
Adding Oauth2 plugin to a specific API

### ISSUE THREAD

**Issue by saithala** [Mashape/kong#1698]
### Summary

I added a OAuth2 plugin to an API. However I need to add another API but do not want that API to have Oauth2 authentication. How can i achieve that?
### Steps To Reproduce
1. Add one API. 
   curl -i -X POST \
   --url http://host:8001/apis/ \
   --data 'name=api1' \
   --data 'upstream_url=http://edge:8765/' \
   --data 'request_host=api2'
2. Add OAuth2 plugin to the api added in step 1
   curl -X POST http://cmcddmz010:8001/apis/api1/plugins --data "name=oauth2"  --data "config.enable_client_credentials=true" 
3. add another api
   curl -i -X POST \
     --url http://host:8001/apis/ \
     --data 'name=api2' \
     --data 'upstream_url=http://edge:8765/' \
4. When I try to access Api2, it rejects the request and asks for Oauth2 token. What Im I doing wrong?
### Additional Details & Logs
- Kong version (`$ kong version`)
- Kong debug-level startup logs (`$ kong start --vv`)
- Kong error logs (`<KONG_PREFIX>/logs/error.log`)
- Kong configuration (registered APIs/Plugins & configuration file)
- Operating System

**Comment by thefosk** [comment]
When you add a plugin to an API, only that API will have the plugin.

**Comment by saithala** [comment]
@thefosk, 
Thanks but I think in my case it is not working that way. I have configured two APIs both with request_path. So to get an oauth token, i have to append the request path to the oauth token uri (apiuri/oauth2/token). I added Oauth2 plugins for both the APIs. But for the third API, I need that to be unauthenticated and did not the oauth2 plugin. I access this third api and it throws an error 
{"error_description":"The access token is missing","error":"invalid_request"}.  Thoughts?

Also another related question is for APIs with request_path, the oauth token URI needs to be appended with the the same request_path.. This causes us to have one token url (appended with appropriate api request_path) for each API. how can we avoid this?
